Gems/ROS2/Code/Source/Camera/ROS2CameraSensorComponent.h

```diff
@@ -40,34 +41,36 @@ namespace ROS2
         static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
         static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
         ~ROS2CameraSensorComponent() override = default;
-        AZ_COMPONENT(ROS2CameraSensorComponent, "{3C6B8AE6-9721-4639-B8F9-D8D28FD7A071}", ROS2SensorComponent);
+
+        AZ_COMPONENT(ROS2CameraSensorComponent, "{3C6B8AE6-9721-4639-B8F9-D8D28FD7A071}", SensorBaseType);
         static void Reflect(AZ::ReflectContext* context);
 
         // AzToolsFramework::Components::EditorComponentBase overrides ..
         void Activate() override;
         void Deactivate() override;
 
         // CameraCalibrationRequestBus::Handler overrides ...
-        AZ::Matrix3x3 GetCameraMatrix() const override;
-        int GetWidth() const override;
-        int GetHeight() const override;
-        float GetVerticalFOV() const override;
+        [[nodiscard]] AZ::Matrix3x3 GetCameraMatrix() const override;
```

Comment:
I would not omit [[nodiscard]] since it is part of CameraCalibrationRequestBus::Handler. It should be called directly from ebus system. I cannot see such a pattern used in o3de source code, so we are probably challenging other compilers (GCC, VS here). We should not use nodiscard at all for events in all this contribution.
I think it will cause some problems in future.

Reply:
Fair point, I removed this from public interface.